src/providers/slot/slot.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@angular/core';
+import { DeepDiff } from 'deep-diff';
+import { flatten } from 'lodash';
+import * as moment from 'moment';
+
+import { SlotItem } from '../slot-selector/slot-item';
+import { ExaminerWorkSchedule } from '../../common/domain/DJournal';
+
+@Injectable()
+export class SlotProvider {
```

Comment:
`SlotProvider` as a name is perhaps a bit generic, could it be something a bit more descriptive like `SlotUtilitiesProvider`?